src/server.cpp

```diff
@@ -36,19 +36,19 @@ void printPacket(const Packet& p)
         std::cout << "\t" << ap->name() << ": " << ap->value() << "\n";
 }
 
-Server::Server(boost::asio::io_service& io_service)
+Server::Server(boost::asio::io_service& io_service, const std::string& secret)
       : m_socket(io_service, udp::endpoint(udp::v4(), 9999))
 {
-    startReceive();
+    startReceive(secret);
```

Comment:
If you put `secret` into a local class member (say, `m_secret`), you won't need to pass it to member functions (`startReceive` and `handleReceive`).